convert2rhel/subscription.py

```diff
@@ -126,13 +126,13 @@ def register_system():
 def get_registration_cmd():
     """Build a command for subscription-manager for registering the system."""
     loggerinst.info("Building subscription-manager command ... ")
-    registration_cmd = "subscription-manager register --force"
+    registration_cmd = ["subscription-manager", "register", "--force"]
     if tool_opts.activation_key:
         # Activation key has been passed
         # -> username/password not required
         # -> organization required
         loggerinst.info("    ... activation key detected: %s" % tool_opts.activation_key)
-        registration_cmd += ' --activationkey="%s"' % tool_opts.activation_key
+        registration_cmd.append("--activationkey=%s" % tool_opts.activation_key)
```

Comment:
The parameter is missing the quotes for %s

```suggestion
        registration_cmd.append('--activationkey="%s"' % tool_opts.activation_key)
```

Reply:
I don't think quotes are needed.  When we pass in a string and then split it into arguments using `shlex.split()` we have to be careful that anything that could contain spaces is enclosed in quotes.  Otherwise `shlex.split()` will split it into two arguments.  When we pass a list of arguments, instead, no additional splitting will occur, each element of the list will be a single argument:
``` python
>>> import subprocess, shlex
>>> open('test case', 'w').close()
>>> # Oops, we forgot to quote the filename
>>> subprocess.Popen(shlex.split('cat test case')).wait()
cat: test: No such file or directory
cat: case: No such file or directory
1
>>> # This is how we have to construct the command as a string:
>>> subprocess.Popen(shlex.split('cat "test case"')).wait()
0
>>> # When we use a list, we don't have to quote:
>>> subprocess.Popen(['cat', 'test case']).wait()
0
```
Not having to quote is part of why this is a more secure API for this function.  It's easy to forget to quote something when constructing a string to pass in (especially when dealing with variables). The code will then work until someone constructs a string with a space in it (possibly maliciously).  Passing a list instead means that we do not have to remember to quote.

Reply:
I agree that the quotes are not needed. Frankly I needed to try it out myself with the _=_ sign to verify it 🤫 The following Popen call:
```python
>>> subprocess.Popen(['subscription-manager', 'register', '--serverurl=host.example.com; rm file']).wait()
Unable to reach the server at host.example.com; rm file:443/subscription
69
```
... is the same as if running the following in shell:
```shell
# subscription-manager register "--serverurl=host.example.com; rm file"
Unable to reach the server at host.example.com; rm file:443/subscription
```

The following code construct would not make me think twice about the functionality. But that would change the command called and that's I believe not the intent of this PR.
```suggestion
        registration_cmd.extend(["--activationkey", "%s" % tool_opts.activation_key])
```

Reply:
Oh, that's cool! Thought that we had to preserve the quotes for these cases (Imagined that this could cause some unreliable calls to subscription-manager). Either way, the other comments about quotes can be ignored safely then!